src/puppetlabs/trapperkeeper/services/scheduler/scheduler_core.clj

```diff
@@ -55,6 +56,26 @@
       ; this can occur if the interface is being used while the scheduler is shutdown
       (log/error e (i18n/trs "Failed to schedule job")))))
 
+(defn cron
+ [cron-string f ^Scheduler scheduler group-name]
+  (try
+    ;; throws parse error if cron-string is invalid
+    (CronExpression/validateExpression cron-string)
+    (let [cron-schedule (CronScheduleBuilder/cronSchedule cron-string)
+          job-name (Key/createUniqueName group-name)
+          job (build-executable-job f job-name group-name)
+          trigger (-> (TriggerBuilder/newTrigger)
+                      (.withSchedule cron-schedule)
+                      (.build))]
+      (.scheduleJob scheduler job trigger)
+      (.getJobKey trigger))
+    (catch SchedulerException e
+      ; this can occur if the interface is being used while the scheduler is shutdown
+      (log/error e (i18n/trs "Failed to schedule job")))
+    (catch java.text.ParseException  e
+      ;; this occurs when the cron-string is invalid
+      (throw (IllegalArgumentException. ^String (i18n/trs "Invalid cron expression"))))))
```

Comment:
There is a form of the IllegalArgumentException that accepts a `Throwable` as a second argument.  https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html#IllegalArgumentException-java.lang.String-java.lang.Throwable-  I'd suggest passing `e` in there to allow the caller to potentially get more information about the parsing exception.